golem/event_bus/base.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
+from typing import Awaitable, Callable, Generic, Optional, Type, TypeVar
+
+from golem.exceptions import GolemException
+
+TCallbackHandler = TypeVar("TCallbackHandler")
+
+TEvent = TypeVar("TEvent", bound="Event")
+
+
+class Event(ABC):
+    """Base class for all events."""
+
+
+class EventBusError(GolemException):
+    pass
+
+
+class EventBus(ABC, Generic[TCallbackHandler]):
+    @abstractmethod
+    async def start(self) -> None:
+        ...
+
+    @abstractmethod
+    async def stop(self) -> None:
+        ...
+
+    @abstractmethod
+    def is_started(self) -> bool:
+        ...
+
+    @abstractmethod
+    async def on(
```

Comment:
@approxit just out of curiosity, why do `on` and `on_once` have to be async? there doesn't seem to be a potential for anything inherently asynchronous to be happening while a handler is being registered....

Reply:
We decided that 'async' will support wider range of possible implementation when waiting for confirmation that callback was register might take a IO operation that would be better to handle asynchronously.